### PR TITLE
[Feature] Add setting to Truncate Video title on Player

### PR DESCRIPTION
**Pull request by Jacekun**
Feature requested on discord.
Ready for merge.

### CODE REVIEW

File: app/src/main/java/com/lagradost/cloudstream3/ui/player/GeneratorPlayer.kt [hunk 1]
```diff
@@ -521,10 +522,11 @@ class GeneratorPlayer : FullScreenPlayer() {
                 tvType = meta.tvType
             }
         }
-
-        player_episode_filler_holder?.isVisible = isFiller ?: false
-
-        player_video_title?.text = if (headerName != null) {
+        //Get limit of characters on Video Title
+        val settingsManager = PreferenceManager.getDefaultSharedPreferences(requireContext())
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/ui/player/GeneratorPlayer.kt:526]: do not use requireContext(), IT WILL crash

Comment (LagradOst) [reply app/src/main/java/com/lagradost/cloudstream3/ui/player/GeneratorPlayer.kt:526]: use context?.let or smth

File: app/src/main/res/xml/settings.xml [hunk 2]
```diff
@@ -20,6 +20,10 @@
                 android:key="@string/quality_pref_key"
                 android:title="@string/watch_quality_pref"
                 android:icon="@drawable/ic_baseline_hd_24" />
+        <Preference
+            android:key="@string/prefer_limit_title_key"
+            android:title="@string/limit_title"
+            android:icon="@drawable/ic_baseline_letter_a" />
```
Comment (LagradOst) [inline app/src/main/res/xml/settings.xml:26]: DO NOT USE .PNG use vector images instead, check out https://fonts.google.com/icons?selected=Material+Icons or use the built in vector image in android studio

File: app/src/main/res/values/array.xml [hunk 3]
```diff
@@ -59,6 +59,21 @@
         <item>3</item>
     </array>
 
+    <array name="limit_title_pref_names">
+        <item>None</item>
```
Comment (LagradOst) [inline app/src/main/res/values/array.xml:63]: There is a  @ string/none

**Review by LagradOst — CHANGES_REQUESTED** [review]
I do not understand why you want this option. If you want to add it I would also suggest an option to simply turn of the title and inf title length. Also check out android:maxLength="[PUT VALUE HERE]"

**Comment by Jacekun** [comment]
> I do not understand why you want this option. If you want to add it I would also suggest an option to simply turn of the title and inf title length. Also check out android:maxLength="[PUT VALUE HERE]"

I personally isn't bother by long titles. This feature is requested by someone on discord and I thought "Why not?" since its possible to be implemented. Also, I'd rather have the setting decide the length than a hardcoded value so users can control how long the title should be on the player UI.

**Comment by LagradOst** [comment]
No, I meant that you can use `.filters = arrayOf(InputFilter.LengthFilter(length))` and android will truncate the text. Also what do you think about adding the option of not display the title, or is that not necessary?

File: app/src/main/res/drawable/ic_baseline_text_format_24.xml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="24dp"
+    android:height="24dp"
+    android:viewportWidth="24"
+    android:viewportHeight="24"
+    android:tint="?attr/colorControlNormal">
```
Comment (LagradOst) [inline app/src/main/res/drawable/ic_baseline_text_format_24.xml:6]: ?attr/white

**Comment by MrConqueror** [comment]
What a useless feature, why not remove Video info quality. Ask shivam for help to do it. Because he did it on his app. We've been asking this so we thought devs of this app can't do it. That's why they're ignoring this feature

**Pull request merged**